stl/inc/ranges

```diff
@@ -7656,6 +7662,401 @@ namespace ranges {
+                return _Sentinel<false>{_Zip.end()};
+            }
+        }
+
+        constexpr auto end() const noexcept(_Is_end_noexcept<true>) // strengthened
+            requires _Enable_const_begin_end
+        {
+            if constexpr (common_range<const _Inner_view>) {
+                return _Iterator<true>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<true>{_Zip.end()};
+            }
+        }
+
+        constexpr auto size() noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<_Inner_view>
+        {
+            return _Zip.size();
+        }
+
+        constexpr auto size() const noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<const _Inner_view>
+        {
+            return _Zip.size();
+        }
+    };
+
+    template <class _Func, class... _Ranges>
+    zip_transform_view(_Func, _Ranges&&...) -> zip_transform_view<_Func, views::all_t<_Ranges>...>;
+
+    namespace views {
+        struct _Zip_transform_fn {
+        private:
+            template <class _Func, class... _Types>
+            static constexpr bool _Is_invocation_noexcept = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    // equivalent to:
+                    // using _Decayed_func = decay_t<_Func>;
+                    //
+                    // return noexcept(((void) _STD declval<_Func&>(),
+                    //     auto(empty<decay_t<invoke_result_t<_Decayed_func&>>>)));
+                    return true;
+                } else {
+                    return noexcept(zip_transform_view(_STD declval<_Func>(), _STD declval<_Types>()...));
+                }
+            }();
+
+            template <class _Func, class... _Types>
+            static constexpr bool _Enable_cpo = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    using _Decayed_func = decay_t<_Func>;
+
+                    if constexpr (move_constructible<_Decayed_func> && regular_invocable<_Decayed_func&>) {
+                        return !is_void_v<invoke_result_t<_Decayed_func&>>;
+                    } else {
+                        return false;
+                    }
+                } else {
+                    return requires(_Func && _Function, _Types && ... _Args) {
+                               zip_transform_view(_STD forward<_Func>(_Function), _STD forward<_Types>(_Args)...);
+                           };
+                }
+            }();
+
+        public:
+            template <class _Func, class... _Types>
+                requires _Enable_cpo<_Func, _Types...>
+            _NODISCARD constexpr auto operator()(_Func&& _Function, _Types&&... _Args) const
+                noexcept(_Is_invocation_noexcept<_Func, _Types...>) // strengthened
+            {
+                if constexpr (sizeof...(_Types) == 0) {
+                    using _Decayed_func = decay_t<decltype((_Function))>;

```

Comment:
The Standard says "let `FD` be `decay_t<decltype((F))>`" because it's phrased in terms of a subexpression `F` whose type it doesn't already know. Here we have `_Func&& _Function`, so we already know the type. (And, notably, `(_Function)` is always an lvalue, unlike `(F)`, so there's already a semantic difference.) Ultimately we're passing the result to `decay_t` so it doesn't really matter, but @CaseyCarter and I think it would be simpler to say `decay_t<_Func>` - and better for throughput.